services/slam/internal/grpchelper/grpchelper.go

```diff
@@ -0,0 +1,29 @@
+// Package grpchelper implements helper functions to be used with slam service grpc clients
+package grpchelper
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	pb "go.viam.com/api/service/slam/v1"
+)
+
+// HelperGetInternalStateCallback helps a client request the internal state stream from a SLAM server.
+func HelperGetInternalStateCallback(ctx context.Context, name string, slamClient pb.SLAMServiceClient) (func() ([]byte, error), error) {
+	req := &pb.GetInternalStateStreamRequest{Name: name}
+
+	resp, err := slamClient.GetInternalStateStream(ctx, req)
+	if err != nil {
+		return nil, errors.Wrap(err, "error getting the internal state from the SLAM client")
+	}
+
+	f := func() ([]byte, error) {
+		chunk, err := resp.Recv()
+		if err != nil {
```

Comment:
This needs to be changed to consider an err.EOF the signal that the response has eneded. 

I'm not 100% sure if when you get an EOF you are guaranteed chunk is nil / empty or not (this needs to be tested), but I'm sure that err.EOF should be the signal we use to determine when all the content has been consumed from resp.Recv()

Reply:
We can have a check for EOF and if triggered we forward the EOF along with the chunks. If its truly empty then appending it shouldn't be an issue to append them before the return in EOF

Reply:
EOF is never sent with a chunk, as seen with integration tests

Reply:
Great, I wasn't sure if you had written your comment before or after confirming that

Reply:
I'm good with keeping this code as is.

The GRPC docs also appear to show that if EOF is returned, the chunk can be safely ignored:
https://grpc.io/docs/languages/go/basics/#server-side-streaming-rpc-1

I'm going to resolve this thread.

Reply:
Also thanks for verifying the behavior!